subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/TestReport.java

```diff
@@ -65,16 +64,7 @@ protected ObjectFactory getObjectFactory() {
      * Returns the directory to write the HTML report to.
      */
     @OutputDirectory
-    public File getDestinationDir() {
-        return destinationDir;
-    }
-
-    /**
-     * Sets the directory to write the HTML report to.
-     */
-    public void setDestinationDir(File destinationDir) {
-        this.destinationDir = destinationDir;
-    }
+    public abstract DirectoryProperty getDestinationDir();
```

Comment:
We need to keep the old getter/setter, but we could introduce a new property `getDestinationDirectory` to replace it.

We could make the old APIs nag

Reply:
Nagging complete, with test coverage.